services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -548,26 +580,30 @@ async def copy_file_s3_s3(self, user_id: str, dest_uuid: str, source_uuid: str):
     async def copy_file_s3_datcore(
         self, user_id: str, dest_uuid: str, source_uuid: str
     ):
+        session = get_client_session(self.app)
+
         # source is s3, get link and copy to datcore
         bucket_name = self.simcore_bucket_name
         object_name = source_uuid
         filename = source_uuid.split("/")[-1]
-        tmp_dirpath = tempfile.mkdtemp()
-        local_file_path = os.path.join(tmp_dirpath, filename)
-        url = self.s3_client.create_presigned_get_url(bucket_name, object_name)
-        session = get_client_session(self.app)
-        async with session.get(url) as resp:
-            if resp.status == 200:
-                f = await aiofiles.open(local_file_path, mode="wb")
-                await f.write(await resp.read())
-                await f.close()
-                # and then upload
-                await self.upload_file_to_datcore(
-                    user_id=user_id,
-                    local_file_path=local_file_path,
-                    destination_id=dest_uuid,
-                )
-        shutil.rmtree(tmp_dirpath)
+
+        s3_dowload_link = self.s3_client.create_presigned_get_url(
+            bucket_name, object_name
+        )
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # FIXME: connect download and upload streams
```

Comment:
Was this done like this to avoid using too much memory with streams?
This is also quite slow, and would benefit a lot if it would be chunked from one request to another. But I guess this would come when this service will be redesigned.

Reply:
Yes, this is what I meant in the FIXME note.
Take into account that storage is a very old code. I had to put this part of the code aside so it could be testable ( I needed to mock downloading from DATCore service).
I did not want to add more changes **in this PR** than the strictly related to the subject of the PR ... so we do not loose focus on the fix we are working here.

In any case, since this is isolated and you insist, I will add it now.

Reply:
I totally agree with you. BUT, this is a small change to the original code ... I did not want to improve it further in this PR since it is NOT related to access-rights (the main topic of this PR).